### PR TITLE
Please work(Party Finder Stats)

### PR DESCRIPTION
**Pull request by TakoTheDev**
If this doesnt work, i am giving up

### CODE REVIEW

File: src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt [hunk 1]
```diff
@@ -46,6 +61,95 @@ class ChatListener {
...
+                    }
+                    return
+                }
+            }
+            val joinedd = dungeon_finder_join_pattern.matcher(unformatted)
+            if (joinedd.find()) {
+                val username = joinedd.group(1)
+                if(username.equals(mc.thePlayer.name)) {
+                    try {
+                        Thread.sleep(1000)
```
Comment (Sychic) [inline src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:114]: This is on the main thread, you should not be calling thread.sleep here.

File: src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt [hunk 2]
```diff
@@ -211,6 +315,262 @@ class ChatListener {
...
+                        } else if (itemName.contains("Axe of the Shredded")) {
+                            hasAots = true
+                        } else if (itemName.contains("Giant's Sword")) {
+                            hasGiantSword = true
+                        } else if (itemName.contains("Shadow Fury")) {
+                            hasShadowFury = true
+                        } else if (itemName.contains("Flower of Truth")) {
+                            hasFlowerOfTruth = true
+                        }
+                    }
```
Comment (Sychic) [inline src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:451]: This should be a when block

File: src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt [hunk 3]
```diff
@@ -211,6 +315,262 @@ class ChatListener {
...
+                            append("§2Giant's Sword: §a§l●\n")
+                        } else if(hasAots) {
+                            append("§aAots: §a§l●\n")
+                        } else if(hasShadowFury) {
+                            append("§8Shadow Fury: §a§l●\n")
+                        } else if(hasFlowerOfTruth) {
+                            append("§cFot: §a§l●\n")
+                        } else {
+                            append("§2Giant's Sword: §c§l●\n")
+                        }
```
Comment (Sychic) [inline src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:487]: This should be a when block

**Comment by TakoTheDev** [comment]
Doesnt work since i could finally test it after downgrading to pre2

**Review by Sychic — CHANGES_REQUESTED** [review]

**Pull request closed without merging**